Libraries/Components/MaskedView/MaskedViewIOS.ios.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule MaskedViewIOS
+ * @flow
+ */
+
+const React = require('React');
+const StyleSheet = require('StyleSheet');
+const View = require('View');
+const ViewPropTypes = require('ViewPropTypes');
+const findNodeHandle = require('ReactNative').findNodeHandle;
+const requireNativeComponent = require('requireNativeComponent');
+const ReactPropTypes = React.PropTypes;
+
+import type { StyleObj } from 'StyleSheetTypes';
+
+/**
+ * Renders the child view with a mask specified in the `renderMask` prop.
+ *
+ * ```
+ * import React from 'react';
+ * import { MaskedView, Text, View } from 'react-native';
+ *
+ * class MyMaskedView extends React.Component {
+ *   render() {
+ *     return (
+ *       <MaskedView
+ *         style={{ flex: 1 }}
+ *         renderMask={() =>
+ *           <View style={styles.maskContainerStyle}>
+ *             <Text style={styles.maskTextStyle}>
+ *               Basic Mask
+ *             </Text>
+ *           </View>
+ *         }
+ *       >
+ *         <View style={{ flex: 1, backgroundColor: 'blue' }} />
+ *       </MaskedView>
+ *     );
+ *   }
+ * }
+ * ```
+ *
+ * The above example will render a view with a blue background that fills its
+ * parent, and then mask that view with text that says "Basic Mask".
+ *
+ * The alpha channel of the view rendered by the `renderMask` prop determines how
+ * much of the view’s content and background shows through. Fully or partially
+ * opaque pixels allow the underlying content to show through but fully
+ * transparent pixels block that content.
+ *
+ */
+class MaskedViewIOS extends React.Component {
+  props: {
+    style?: StyleObj,
+    /**
+     * `MaskedView` only accepts a single child as a view to be masked.
+     */
+    children: ReactElement<*>,
+    /**
+     * Should return a React element to be rendered and applied as the
+     * mask for the child element.
+     */
+    renderMask: () => React.Element<*>,
+  };
+
+  state: {
+    maskViewNodeHandle: ?mixed,
+  } = {
+    maskViewNodeHandle: null,
+  };
+
+  static propTypes = {
+    ...ViewPropTypes,
+    renderMask: ReactPropTypes.func.isRequired,
+  };
+
+  _hasWarnedInvalidChildren = false;
+  _hasWarnedInvalidRenderMask = false;
+
+  render() {
+    if (!this.props.children || React.Children.count(this.props.children) > 1) {
+      if (!this._hasWarnedInvalidChildren) {
+        console.warn(
+          'MaskedView: MaskedView requires a single child React Element.'
+        );
+        this._hasWarnedInvalidChildren = true;
+      }
+      return null;
+    }
+
+    const target = React.Children.only(this.props.children);
+
+    if (typeof this.props.renderMask !== 'function') {
```

Comment:
You could -- this method gives a better warning though, and is a convention I've seen followed elsewhere in the codebase.